Assets/Scripts/Interactables/CocktailFridge.cs

```diff
@@ -1,13 +1,102 @@
+using UnityEngine;
+using System.Linq;
+
 namespace FMS.TapperRedone.Interactables
 {
-	public class CocktailFridge : InteractableObject
-	{
-		public Inventory.Item itemToSpawn;
-
-		public override void Interact(Characters.PlayerInteraction player)
-		{
-			base.Interact(player);
-			player.CurrentInventory.AddItem(itemToSpawn);
-		}
-	}
+    public class CocktailFridge : InteractableObject
+    {
+        public Inventory.Item[] Ingredients; //array to store colour coded ingredients items as ItemType
+        private SpriteRenderer FridgeSpriteRenderer; // Reference to the SpriteRenderer of the fridge
+
+        private enum ColorCategory //short enum list for the colour window of error
+        {
+            Red,
+            Yellow,
+            Blue,
+            Unknown
+        }
+
+        public void Awake()
+        {
+            if (FridgeSpriteRenderer == null)
+            {
+                //getting the sprite renderer of assigned object prefab 
+                FridgeSpriteRenderer = GetComponent<SpriteRenderer>();
+            }
+        }
+
+        public override void Interact(Characters.PlayerInteraction player)
+        {
+            base.Interact(player);
+           
+
+            if (FridgeSpriteRenderer != null)
+            {
+                Color fridgeColor = FridgeSpriteRenderer.color;
+                Inventory.Item itemToSpawn = GetItemForColor(fridgeColor);
+
+                if (itemToSpawn != null)  //debugging to confirm if colour exists
+                {
+                    player.CurrentInventory.AddItem(itemToSpawn);
+                }
+                else
+                {
+                    Debug.LogWarning("No item found for the color: " + fridgeColor);
+                }
+            }
+            else
+            {
+                Debug.LogError("SpriteRenderer component is missing!");
+            }
+        }
+
+        private Inventory.Item GetItemForColor(Color color) //which ingredient to give to player based on the fridge colour
+        {
+            switch (GetColorCategory(color))
+            {
+                case ColorCategory.Red:
+                    return GetIngredientByType(Inventory.Item.ItemType.RedIngredient);
+
+                case ColorCategory.Blue:
+                    return GetIngredientByType(Inventory.Item.ItemType.BlueIngredient);
+               
+                case ColorCategory.Yellow:
+                    return GetIngredientByType(Inventory.Item.ItemType.YellowIngredient);
+                default:
+                    return null;
+            }
+
+        }
+
+        private ColorCategory GetColorCategory(Color color) //maps the Color to a ColorCategory enum predifined
+        {
+            // Here we compare colors with a small tolerance for floating-point precision issues
+            if (ColorsAreClose(color, new Color(0.934f, 0f, 0f, 1f)))
```

Comment:
Set these as constants, either in code (final or static), or through some kind of prefab.